components/Dialogs/tests/unit/MDCAlertControllerLayoutTests.m

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialDialogs.h"
+
+#import "MDCAlertControllerView+Private.h"
+
+#import <XCTest/XCTest.h>
+
+static const CGFloat kFourInchPortraitWidth = 320.0;
+static const CGFloat kFourInchLandscapeWidth = 568.0;
+
+@interface MDCAlertControllerLayoutTests : XCTestCase
+
+@end
+
+@implementation MDCAlertControllerLayoutTests {
+  MDCAlertController *_alertController;
+}
+
+- (void)setUp {
+  [super setUp];
+}
+
+// The large point size Urdu text in this dialog should lay out the two buttons vertically on
+// a 4" portrait size screen and horizontally in landscape
+- (void)setupLargePointSizeUrduNetworkRetryDialog {
+  NSString *networkFailureInUrdu = @"براہ کرم اپنا نیٹ ورک کنکشن چیک کریں اور دوبارہ کوشش کریں۔";
+  _alertController = [MDCAlertController alertControllerWithTitle:nil message:networkFailureInUrdu];
+  NSString *urduFontName = @"NotoNastaliqUrdu";
+  UIFont *dialogBodyFont;
+  UIFont *dialogButtonFont;
+  if (@available(iOS 11, *)) {
+    // Noto Nastaliq Urdu was added in iOS 11, and is an extremely tall
+    // font for any given nominal point size.
+    dialogBodyFont = [UIFont fontWithName:urduFontName size:20.0];
+    dialogButtonFont = [UIFont fontWithName:urduFontName size:26.0];
+  } else {
+    dialogBodyFont = [UIFont systemFontOfSize:20.0];
+    dialogButtonFont = [UIFont systemFontOfSize:26.0];
```

Comment:
Shouldn't we test other tall fonts in non ios11 versions as well?

Reply:
I think that would be ideal. `NotoNastaliqUrdu` is the known tall font that was introduced and broke the functionality, I think it is fine to leave it for pre iOS 11 until we get more information on any tall font pre iOS 11 that breaks the dialog.